### ISSUE TITLE
210,139 valid NDC concepts have no relationship

### ISSUE THREAD

**Issue by schuemie** [OHDSI/Vocabulary-v5.0#557]
I found many NDC concepts that are valid, yet have no relationship whatsoever, including any mapping to standard concepts:

```sql
SELECT COUNT(*)
FROM concept
LEFT JOIN concept_relationship
  ON concept_id = concept_id_1
WHERE concept_id_2 IS NULL
  AND concept.invalid_reason IS NULL
  AND vocabulary_id = 'NDC';
```

Many of these are actually used (e.g. in Synpuf), for example [this one](https://athena.ohdsi.org/search-terms/terms/44876855) or [this one](https://athena.ohdsi.org/search-terms/terms/44911308).

**Comment by cgreich** [comment]
You are correct, @schuemie. They are coming from some less than well organized source and need to be run through the RxNorm Extension machinery (the "boiler"). 

Where are we on that, @mik-ohdsi?

**Comment by mik-ohdsi** [comment]
Hi @cgreich and @schuemie ,
currently we are investigating how to make better use of the the RxNorm API to get historic NDC codes, which probably will account at least for a good part of these. Having those should give us links to rxcui entries and allow mapping. 

However, among those NDCs that have no mappings, we also find lots of non-drugs or cosmetic products:


concept_id | concept_name | concept_class_id
-- | -- | --
36490762 | DERMA SOLEIL NATURAL SUNBLOCK SPF 30 - octocrylene, octyl methoxycinnamate, octyl salicylate, oxybenzone lotion | 11-digit NDC
44483247 | ketorolac tromethamine 30mg/mL INTRAMUSCULAR; INTRAVENOUS INJECTION, SOLUTION | 9-digit NDC
36008728 | hydrocortisone 0.5% / hydroquinone 8% / tretinoin 0.025% 8g/100g / .5g/100g / .025g/100g TOPICAL EMULSION | 11-digit NDC
45046517 | Salicylic Acid 20 mg/mL LIQUID [Neutrogena Oil Free Acne Stress Control] | 9-digit NDC
45826314 | titanium dioxide 45mL/100mL TOPICAL LIQUID [almost makeup spf 15] | 9-digit NDC
46346849 | CLE DE PEAU BEAUTE RADIANT FLUID FOUNDATION I10 - octinoxate and titanium dioxide cream | 11-digit NDC
44897721 | PNV-DHA PLUS SOFTGEL | 11-digit NDC
35405096 | DIGITALIS PURPUREA 200C - digitalis purpurea pellet | 11-digit NDC

From the random first 8 entries, 2 seem to be actual drugs and the remaining 6 are not something we could successfully map.

Having said that: let me ping @p-talapova who is assigned to review our options with the RxNorm API.

**Comment by MeliHenry** [comment]
Hi @mik-ohdsi, do we have an update on this now?
